### PR TITLE
Add more SCE contents to merged objects page

### PR DESCRIPTION
**Pull request by sjspielman**
Towards #219 

This PR adds more towards merged _SCE_ file contents. I added the following sections:

- `SingleCellExperiment cell metrics`
- `SingleCellExperiment gene information and metrics`
- `SingleCellExperiment experiment metadata`

The content mostly mirrors what's in `sce_file_contents.md`, but with some wording changes. Let me know what you think of phrasing throughout and if it all makes sense, and also whether we might want additional organization/subsections for the metadata.

### CODE REVIEW

File: docs/merged_objects.md
Comment (sjspielman) [reply docs/merged_objects.md:76]: argh foiled by copy/paste 😬

Comment (sjspielman) [reply docs/merged_objects.md]: Ok, I've gotten all the `colData` here: https://github.com/AlexsLemonade/scpca-docs/pull/237/commits/7c7861eb87e71cd94b4cc9be410a02b86d3ece3e

And found something else we need to fix! Looks like merged objects are missing `scpca_filter` and `adt_scpca_filter` (if at least 1 library ahs cite). I feel pretty sure we want to keep those columns around! I'll file a PR in `scpca-nf` to retain those columns.

Comment (sjspielman) [reply docs/merged_objects.md]: Note, currently updating colData bits to properly handle multiplex which won't have this (related to https://github.com/AlexsLemonade/scpca-docs/pull/237#discussion_r1457526408)

**Comment by sjspielman** [comment]
I think this is now at a point where it's worth another look. I'm particularly interested in your thoughts on how I've presented the multiplex sample metadata here. Originally I was going to hold off on those until a future PR where I plan to add "additional considerations for multiplex" section, but given the different organization of the merged object compared to individual libraries, I think some of the way I've done it makes sense? Let me know if you agree or not!

I also found some things to fix along the way in `sce_file_contents`, including adding `gene_ids` as a rowData column, and updating a lot of the spots where we say "only present in filtered" -> "only present in filtered and processed".

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:53]:
```suggestion
This `DataFrame` also contains additional sample metadata information stored in the `colData` slot `DataFrame` for all projects that do not contain multiplexed libraries.
Read more about the sample metadata that is included in the [`Sample metadata` section](#singlecellexperiment-sample-metadata), 
```
Since the presence of sample metadata is dependent on any libraries in a project being multiplexed, let's specify that.

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:96]:
```suggestion
However, there will only be a single `gene_symbol` and `gene_id` column, as this information equally pertains to all libraries.
```
Is it `gene_id` or `gene_ids`, I think `gene_id`?

File: docs/merged_objects.md
Comment (allyhawkins) [reply docs/merged_objects.md:121]: I think I would add a note here too that any projects with multiplexed libraries will contain an additional `sample_metadata` data frame and link to the section below.

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:151]: I would remove every reference to `Only present for x objects`. There's only one type of merged object.

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:182]:
```suggestion
If the project does_not_ contain multiplexed libraries, this information can be found as additional columns in the `colData` slot's `DataFrame`, along with [cell metrics](#singlecellexperiment-cell-metrics).
```

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:188]:
```suggestion
If the project contains multiplexed libraries, this information is stored in the `metadata` slot in the `sample_metadata` field as a `data.frame`:
```

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:195]:
```suggestion
Note that, in {ref} `individual library objects<sce_file_contents:singlecellexperiment sample metadata>`, this sample metadata information is instead stored in the `SingleCellExperiment` object's `metadata` slot.
```

**Review by allyhawkins — COMMENTED** [review]
The structure here looks good, but I had some additional comments, mostly wording suggestions.

File: docs/merged_objects.md
Comment (sjspielman) [reply docs/merged_objects.md:97]: Ah, `gene_ids`! I'll add this on my end.

```
> sce <- readRDS("SCPCL000001_processed.rds")
> names(rowData(sce))
[1] "gene_ids"    "gene_symbol" "mean"        "detected"  
```

File: docs/merged_objects.md
Comment (sjspielman) [reply docs/merged_objects.md:152]: 🤦‍♀️

File: docs/merged_objects.md
Comment (sjspielman) [reply docs/merged_objects.md]: I ended up actually adding a row to the table, is that ok?

**Comment by sjspielman** [comment]
Getting closer!!

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:53]:
```suggestion
This `DataFrame` also contains additional sample metadata information stored in the `colData` slot for all projects that do not contain multiplexed libraries.
```
I don't think we need the extra `DataFrame`.

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:120]:
```suggestion
| `sample_id`                    | A vector of sample IDs which are included in the merged object, each in the form  `SCPCS000000`.
```

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:123]: Maybe switch this order? I believe sample metadata should be first.

File: docs/merged_objects.md
Comment (allyhawkins) [inline docs/merged_objects.md:196]: I think I would move this sentence to be right after line 189 and state that just like merged objects with multiplexed libraries, all individual objects will contain the `sample_metadata` in the `metadata` slot.

**Review by allyhawkins — APPROVED** [review]
Some additional minor comments, but I don't need to see this again.

**Comment by sjspielman** [comment]
> I think I would move this sentence to be right after line 189 and state that just like merged objects with multiplexed libraries, all individual objects will contain the sample_metadata in the metadata slot.

@allyhawkins can you quickly peek at my wording for this in https://github.com/AlexsLemonade/scpca-docs/pull/237/commits/857c76f7e2e7ca2318aaf1bec185c75abadb3c82? Is this about what you had in mind?

**Comment by allyhawkins** [comment]
> > I think I would move this sentence to be right after line 189 and state that just like merged objects with multiplexed libraries, all individual objects will contain the sample_metadata in the metadata slot.
> 
> @allyhawkins can you quickly peek at my wording for this in [857c76f](https://github.com/AlexsLemonade/scpca-docs/commit/857c76f7e2e7ca2318aaf1bec185c75abadb3c82)? Is this about what you had in mind?

Yea that looks good!

**Pull request merged**
